Provide rvalue cast operator for ResultValue

afrantzis opened the pull request:
Allows objects returned as ResultValue (instead of ResultValueType::type) to be moved without requiring additional casting or explicit use of ResultValue::value.

Resolves #589

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/KhronosGroup/Vulkan-Hpp?pullRequest=590) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/KhronosGroup/Vulkan-Hpp?pullRequest=590) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/KhronosGroup/Vulkan-Hpp?pullRequest=590) it.</sub>

asuessenbach commented:
This looks like a reasonable addition. Thanks for contributing to this project!
Would you mind, though, to add the generated vulkan.hpp as well?

afrantzis commented:
@asuessenbach Hi! I have updated this MR to include the change to the generated file. I also changed the formatting in the generator string slightly to better match what's mostly used elsewhere (not that it matters much since everything is passed through clang-format).

Do you think it would be worth it to also add a small compilation test case for this behavior? If so, what's the best place for it (tests?, samples?).

Thanks!

asuessenbach commented:
A little compilation test would be great to have!
But I would not call it required.
If you'd like to add such a test, please introduce a new project in the Tests directory.

afrantzis commented:
> If you'd like to add such a test, please introduce a new project in the Tests directory.

Done!

asuessenbach commented:
Thanks a lot for your contribution!
